pkg/controller/machineautoscaler/validator.go

```diff
@@ -0,0 +1,77 @@
+package machineautoscaler
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	autoscalingv1beta1 "github.com/openshift/cluster-autoscaler-operator/pkg/apis/autoscaling/v1beta1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// Validator validates MachineAutoscaler resources.
+type Validator struct {
+	client  client.Client
+	decoder *admission.Decoder
+}
+
+// NewValidator returns a new Validator.
+func NewValidator() *Validator {
+	return &Validator{}
+}
+
+// Validate validates the given MachineAutoscaler resource.
+func (v *Validator) Validate(ma *autoscalingv1beta1.MachineAutoscaler) (bool, utilerrors.Aggregate) {
+	errs := []error{}
```

Comment:
Replace with nil slice declaration? - `var errs []error`

Reply:
I'm happy to, but any reason? Seems half of people prefer one way and the other half the other. Doesn't make any difference to me.

Reply:
I just try to be consistent with the implied semantics, ala:

```go
	var nilSlice []int
	emptySlice1 := make([]int, 0)
	emptySlice2 := []int{}

	fmt.Println(nilSlice == nil)    // true
	fmt.Println(emptySlice1 == nil) // false
	fmt.Println(emptySlice2 == nil) // false
```

Reply:
Not a blocker for me  - just something I try to be self-consistent with and that's why I noticed it.

Reply:
Sure. Makes sense.

Reply:
Fixed.